apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -115,15 +116,34 @@ export const selectSection = sectionId => ({ type: SELECT_SECTION, sectionId });
 export const removeSection = sectionId => ({ type: REMOVE_SECTION, sectionId });
 
 /**
+ * Changes the hidden state of a given section, persisting these changes to the
+ * server
+ * @param {number} sectionId
+ */
+export const toggleSectionHidden = sectionId => (dispatch, getState) => {
+  dispatch(beginEditingSection(sectionId, true));
+  const state = getState();
+  const currentlyHidden = getRoot(state).sections[sectionId].hidden;
+  dispatch(editSectionProperties({hidden: !currentlyHidden}));
+  return dispatch(finishEditingSection());
```

Comment:
I could have built a new API just for toggling hidden on the server, but instead opted to take advantage of the existing section update logic. The one new thing I needed was being able to do an edit without popping our EditSection dialog. I did this by adding a new argument to `beginEditingSection` to allow us to do so silently.